.github/workflows/plyer_deploy.yml

```diff
@@ -0,0 +1,46 @@
+name: Deploy to PyPI
+
+on:
+  create
+
+jobs:
+  deploy:
+    if: startsWith(github.ref, 'refs/tags/')
+
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.7.x'
+
+    - name: Install dependencies
+      run: |
+        source ci/ci_ubuntu.sh
+        deployment_dependencies
+
+    - name: Create artifacts
+      uses: actions/upload-artifact@v1
+      with:
+        name: plyer_artifact
+        path: dist
+
+    - name: Upload to GitHub Releases
+      uses: softprops/action-gh-release@v0.1.5
+
+      env:
+        GITHUB_TOKEN: ${{ secrets.github_release }}
+
+      with:
+        files: dist/*
+        draft: true
+
+    - name: Build and publish to PyPI
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+
+      run: |
+        source ci/ci_ubuntu.sh
+        deploy
```

Comment:
You are running the deploy step after the artifacts and release upload. I think this needs to happen first.

So the order should be: generate deploy artifacts, then upload artifact to job, then release upload, and finally pypi upload. Pypi upload always needs to be last because that is the most likely to fail, so that's why the order is artifact (least likely to fail) and release page (slightly more likely to fail compared to artifact, but less so than pypi) and then pypi.

Reply:
I did it here https://github.com/kivy/plyer/pull/544/commits/1159d9270f668d3b1084e327902e8556be959f9e